rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1242,6 +1246,52 @@ public BooleanWrapper editNodeSource(String nodeSourceName, String infrastructur
         return new BooleanWrapper(true);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public BooleanWrapper updateDynamicParameters(String nodeSourceName, String infrastructureType,
+            Object[] infraParams, String policyType, Object[] policyParams) {
```

Comment:
this method is a bit too complex to follow :(. can we refactor it? 💃

Reply:
I suppose it can be divided into higher level functions

Reply:
Please add comments as well for better understanding

Reply:
Yes, I extracted more methods. Some logic is moved to NodeSourceParameterHelper class.